app_dart/test/request_handlers/get_status_test.dart

```diff
@@ -74,5 +83,65 @@ void main() {
 
       expect(result['AgentStatuses'], equals(expectedOrderedAgents));
     });
+
+    test('reports statuses without input commit key', () async {
+      final Commit commit1 = Commit(
+          key: config.db.emptyKey.append(Commit,
+              id: 'flutter/flutter/ea28a9c34dc701de891eaf74503ca4717019f829'),
+          timestamp: 3);
+      final Commit commit2 = Commit(
+          key: config.db.emptyKey.append(Commit,
+              id: 'flutter/flutter/d5b0b3c8d1c5fd89302089077ccabbcfaae045e4'),
+          timestamp: 1);
+      config.db.values[commit1.key] = commit1;
+      config.db.values[commit2.key] = commit2;
+      buildStatusProvider = FakeBuildStatusProvider(
+          commitStatuses: <CommitStatus>[
+            CommitStatus(commit1, const <Stage>[]),
+            CommitStatus(commit2, const <Stage>[])
+          ]);
+      handler = GetStatus(
+        config,
+        datastoreProvider: () => DatastoreService(db: config.db),
+        buildStatusProvider: buildStatusProvider,
+      );
+
+      final Map<String, dynamic> result = await decodeHandlerBody();
+
+      expect(result['Statuses'].length, 2);
+    });
+
+    test('reports statuses with input commit key', () async {
+      final Commit commit1 = Commit(
+          key: config.db.emptyKey.append(Commit,
+              id: 'flutter/flutter/ea28a9c34dc701de891eaf74503ca4717019f829'),
+          timestamp: 3);
+      final Commit commit2 = Commit(
+          key: config.db.emptyKey.append(Commit,
+              id: 'flutter/flutter/d5b0b3c8d1c5fd89302089077ccabbcfaae045e4'),
+          timestamp: 1);
+      config.db.values[commit1.key] = commit1;
+      config.db.values[commit2.key] = commit2;
+      buildStatusProvider = FakeBuildStatusProvider(
+          commitStatuses: <CommitStatus>[
+            CommitStatus(commit1, const <Stage>[]),
+            CommitStatus(commit2, const <Stage>[])
+          ]);
+      handler = GetStatus(
+        config,
+        datastoreProvider: () => DatastoreService(db: config.db),
+        buildStatusProvider: buildStatusProvider,
+      );
+
+      const String expectedLastCommitKeyEncoded =
+          'ahNzfmZsdXR0ZXItZGFzaGJvYXJkckcLEglDaGVja2xpc3QiOGZsdXR0ZXIvZmx1dHRlci9lYTI4YTljMzRkYzcwMWRlODkxZWFmNzQ1MDNjYTQ3MTcwMTlmODI5DA';
+
+      tester.request = FakeHttpRequest(queryParametersValue: <String, String>{
+        GetStatus.lastCommitKeyParam: expectedLastCommitKeyEncoded,
+      });
+      final Map<String, dynamic> result = await decodeHandlerBody();
+
+      expect(result['Statuses'].length, 1);
```

Comment:
Let's verify that `result['Statuses'].first` is the status for `commit2`.

Reply:
Added.